### PR TITLE
INSTALL: update the list known OSes and CPU archs curl has run on

### PR DESCRIPTION
**Pull request by bagder**

### CODE REVIEW

File: docs/INSTALL.md [hunk 1]
```diff
@@ -447,77 +447,24 @@ line.  Following is a list of appropriate key words:
...
-  - Nios II uClinux
+This is a probably incomplete list of known CPU architectures and operating
+systems that curl has been compiled for. If you know a system curl compiles
+and runs on, that isn't listed, please let us know!
+
+## 79 Operating Systems
+
+AIX, AmigaOS, Android, BeOS, Blackberry 10, Blackberry Tablet OS, Cell OS,
+ChromeOS, Cygwin, Dragonfly BSD, eCOS, FreeBSD, FreeDOS, FreeRTOS, Fuchsia,
+Garmin OS, Genode, Haiku, Hardedened BSD, HP-UX, Hurd, Illumos, Integrity,
```
Comment (dfandrich) [inline docs/INSTALL.md:458]: Typo: Hardened (presumably)

File: docs/INSTALL.md [hunk 2]
```diff
@@ -447,77 +447,24 @@ line.  Following is a list of appropriate key words:
...
+iOS, ipadOS, IRIX, LineageOS, Linux, Mac OS 9, macOS, Mbed, Micrium, MINIX,
+MorphOS, MPE/iX, MS DOS, NCR MP-RAS, NetBSD, Netware, Nintendo Switch,
```
Comment (dfandrich) [inline docs/INSTALL.md:460]: This is more typically MS-DOS.

File: docs/INSTALL.md [hunk 3]
```diff
@@ -447,77 +447,24 @@ line.  Following is a list of appropriate key words:
...
+NonStop OS, NuttX, OpenBSD, OpenStep, OS/2, OS/400, OS21, Plan 9,
+PlayStation Portable, QNX, Qubes OS, ReactOS, Redox, RICS OS, Sailfish OS,
```
Comment (dfandrich) [inline docs/INSTALL.md:462]: Typo: RISC

File: docs/INSTALL.md (hunk 1 above)
Comment (bagder) [reply docs/INSTALL.md:458]: Oops, and people tell me it's actually "HardenedBSD" without any space..

**Pull request closed without merging**